Zend/tests/enum/025.phpt

```diff
@@ -0,0 +1,14 @@
+--TEST--
+Enum unserialize
+--FILE--
+<?php
+
+enum Foo {
+    case Bar;
+}
+
+var_dump(unserialize('E:8:"Foo::Bar";'));
```

Comment:
Would be useful to add a unit test that this properly tracks the reference count if you unserialize more than once (see comment about ZVAL_COPY vs ZVAL_COPY_VALUE)

Could also use `debug_zval_dump()`, both to validate that the `refcount` has the expected value and that the way that `debug_zval_dump` renders enum values is reasonable.

Reply:
Also, does php have any validation that the reference count of class constants is positive during request shutdown (It might, I just don't know if it does)? If not, it may be useful to add for enum classes in debug builds now that `const GLOBAL_OR_CLASS_CONST = Suit::Hearts;` is possible.

Reply:
Nice, this actually caught a bug.

> Also, does php have any validation that the reference count of class constants is positive during request shutdown (It might, I just don't know if it does)

Yes, there already is some logic that checks that the reference count of constants is correct during shutdown.